### PR TITLE
Http resilience readme

### PR DESCRIPTION
**Pull request by Tratcher**
Descriptions borrowed from the API docs

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4663)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.Http.Resilience/README.md [hunk 1]
```diff
@@ -18,6 +18,40 @@ Or directly in the C# project file:
 </ItemGroup>
 ```
 
+## Usage Examples
+
+When configuring an HttpClient through the client factory the following extensions can add a set of pre-configured hedging or resilience behaviors. These pipelines combine multiple strategies with pre-configured defaults.
+- The total request timeout pipeline applies an overall timeout to the execution, ensuring that the request including hedging attempts, does not exceed the configured limit.
+- The retry pipeline retries the request in case the dependency is slow or returns a transient error.
+- The bulkhead pipeline limits the maximum number of concurrent requests being send to the dependency.
```
Comment (martintmk) [inline src/Libraries/Microsoft.Extensions.Http.Resilience/README.md:26]:
```suggestion
- The rate limiter pipeline limits the maximum number of requests being send to the dependency.
```

File: src/Libraries/Microsoft.Extensions.Http.Resilience/README.md [hunk 2]
```diff
@@ -18,6 +18,40 @@ Or directly in the C# project file:
 </ItemGroup>
 ```
 
+## Usage Examples
+
+When configuring an HttpClient through the client factory the following extensions can add a set of pre-configured hedging or resilience behaviors. These pipelines combine multiple strategies with pre-configured defaults.
```
Comment (martintmk) [inline src/Libraries/Microsoft.Extensions.Http.Resilience/README.md:23]:
```suggestion
When configuring an HttpClient through the client factory the following extensions can add a set of pre-configured hedging or resilience behaviors. These pipelines combine multiple resilience strategies with pre-configured defaults.
```

File: src/Libraries/Microsoft.Extensions.Http.Resilience/README.md [hunk 3]
```diff
@@ -18,6 +18,40 @@ Or directly in the C# project file:
...
+```csharp
+var clientBuilder = services.AddHttpClient("MyClient");
+
+clientBuilder.AddStandardHedgingHandler().Configure(o =>
+{
+    o.TotalRequestTimeout.Timeout = TimeSpan.FromSeconds(10);
+});
+```
 
 ## Feedback & Contributing
```
Comment (martintmk) [inline src/Libraries/Microsoft.Extensions.Http.Resilience/README.md:56]: How about adding "Custom resilience" section with the note that if standard hedging and standard resilience are not suitable once can compose their own pipeline?

File: src/Libraries/Microsoft.Extensions.Http.Resilience/README.md (hunk 2 above)
Comment (martintmk) [inline src/Libraries/Microsoft.Extensions.Http.Resilience/README.md:23]:
```suggestion
When configuring an HttpClient through the [HTTP client factory](https://learn.microsoft.com/dotnet/core/extensions/httpclient-factory) the following extensions can add a set of pre-configured hedging or resilience behaviors. These pipelines combine multiple strategies with pre-configured defaults.
```

**Review by martintmk — APPROVED** [review]

File: src/Libraries/Microsoft.Extensions.Http.Resilience/README.md [hunk 4]
```diff
@@ -18,6 +18,55 @@ Or directly in the C# project file:
...
+```
+
+### Custom Resilience
+
+For more granular control a custom pipeline can be constructed.
+
+```csharp
+var clientBuilder = services.AddHttpClient("MyClient");
+
+clientBuilder.AddResilienceHandler("myHandler")
```
Comment (martintmk) [inline src/Libraries/Microsoft.Extensions.Http.Resilience/README.md:63]: This is not the compilable API. Arent you using the R9 API here?

Comment (Tratcher) [reply src/Libraries/Microsoft.Extensions.Http.Resilience/README.md]: I was referencing the wrong branch, fixed.

File: src/Libraries/Microsoft.Extensions.Http.Resilience/README.md [hunk 5]
```diff
@@ -18,6 +18,57 @@ Or directly in the C# project file:
...
+
+For more granular control a custom pipeline can be constructed.
+
+```csharp
+var clientBuilder = services.AddHttpClient("MyClient");
+
+clientBuilder.AddResilienceHandler("myHandler", b =>
+{
+    b.AddConcurrencyLimiter(new ConcurrencyLimiterOptions())
+     .AddTimeout(new TimeoutStrategyOptions())
```
Comment (martintmk) [inline src/Libraries/Microsoft.Extensions.Http.Resilience/README.md:66]:
```suggestion
     .AddTimeout(new HttpTimeoutStrategyOptions())
```

It's better to use http specific options when possible. Applies to other as well. (retries, CB)

File: src/Libraries/Microsoft.Extensions.Http.Resilience/README.md [hunk 6]
```diff
@@ -18,6 +18,57 @@ Or directly in the C# project file:
...
+### Custom Resilience
+
+For more granular control a custom pipeline can be constructed.
+
+```csharp
+var clientBuilder = services.AddHttpClient("MyClient");
+
+clientBuilder.AddResilienceHandler("myHandler", b =>
+{
+    b.AddConcurrencyLimiter(new ConcurrencyLimiterOptions())
```
Comment (martintmk) [inline src/Libraries/Microsoft.Extensions.Http.Resilience/README.md:65]:
```suggestion
    b.AddConcurrencyLimiter(100)
```

The default options won't work so either use this overload or initialize properties explicitly.

File: src/Libraries/Microsoft.Extensions.Http.Resilience/README.md [hunk 7]
```diff
@@ -18,6 +18,57 @@ Or directly in the C# project file:
...
+```
+
+### Custom Resilience
+
+For more granular control a custom pipeline can be constructed.
+
+```csharp
+var clientBuilder = services.AddHttpClient("MyClient");
+
+clientBuilder.AddResilienceHandler("myHandler", b =>
```
Comment (martintmk) [inline src/Libraries/Microsoft.Extensions.Http.Resilience/README.md:63]: This pipeline is strange 😁

Let's put at least timeout at the bottom and fallback at the top. Some required properties are not set so it won't work. I suggest to try to send request once to see what's missing.

File: src/Libraries/Microsoft.Extensions.Http.Resilience/README.md (hunk 5 above)
Comment (Tratcher) [reply src/Libraries/Microsoft.Extensions.Http.Resilience/README.md:66]: How is HttpTimeoutStrategyOptions better when it doesn't have any properties? Why does it even exist?

**Review by martintmk — APPROVED** [review]

**Pull request merged**